src/io/SystemMetadata.tsx

```diff
@@ -0,0 +1,38 @@
+export interface SystemMetadata {
+    cpuUsage: number;
+    memoryUsagePercentage: number;
+    memoryUsed: number;
+    memoryTotal: number;
+    memoryAvailable: number;
+    systemUptime: number;
+}
+
+export class SystemMetadataUtils {
+
+    public static memoryStringFormat(numberToFormat: number) {
+        // number of bytes in a terabyte, gigabyte, and megabyte.
```

Comment:
To improve readability, I'd do two things:
* store each value in a properly named constant, since each of them is used two times
* write the calculation instead of the raw value (everyone knows there are 1024 bytes in a KiB [notice the *i*, I'll get back to it in the comment below], but it's not immediate to remember the larger numbers)

So something like:
```
// (at the top of the file)
const BYTES_IN_KIBIBYTE: number = 1024;
const BYTES_IN_MEBYBYTE: number = BYTES_IN_KIBIBYTE * 1024;
const BYTES_IN_GIBIBYTE: number = BYTES_IN_MEBYBYTE * 1024;
// etc, then in the function use these constants instead of the literal numbers
```